pkg/docker/registry/registry.go

```diff
@@ -57,7 +58,22 @@ var secretAnnotations = map[string]string{
 	"helm.sh/hook-weight":    "-9999",
 }
 
-func GetRegistryProxyInfo(license *kotsv1beta1.License, installation *kotsv1beta1.Installation, app *kotsv1beta1.Application) *RegistryProxyInfo {
+func GetRegistryProxyInfo(license *kotsv1beta1.License, installation *kotsv1beta1.Installation, app *kotsv1beta1.Application) (*RegistryProxyInfo, error) {
+	if util.IsEmbeddedCluster() {
+		registryProxyInfo, err := getEmbeddedClusterRegistryProxyInfo(license)
+		if err != nil {
+			return nil, errors.Wrap(err, "embedded cluster")
+		}
+
+		// getting the registry and proxy from the installation spec takes precedence over the application spec
```

Comment:
i don't think this is true. we shouldn't fall back to anything kots does. if the env vars aren't set, we should fall back to defaults

Reply:
fixed